tests/0137-barrier_batch_consume.c

```diff
@@ -0,0 +1,290 @@
+
+        test_consumer_close(consumer);
+
+        rd_kafka_destroy(consumer);
+
+        SUB_TEST_PASS();
+}
+
+
+static void do_test_consume_batch_with_pause_and_resume() {
+        const int partition_cnt = 2;
+        rd_kafka_queue_t *rkq;
+        const char *topic;
+        rd_kafka_t *consumer;
+        int p;
+        const int timeout_ms = 10000;
+        uint64_t testid;
+        const int consume_msg_cnt = 10;
+        const int produce_msg_cnt = 8;
+        rd_kafka_conf_t *conf;
+        consumer_t consumer_args = RD_ZERO_INIT;
+        test_msgver_t mv;
+        thrd_t thread_id;
+        rd_kafka_resp_err_t err;
+        rd_kafka_topic_partition_list_t *parition_list;
+        int32_t pause_partition = 0;
+
+        SUB_TEST();
+
+        test_conf_init(&conf, NULL, 60);
+        test_conf_set(conf, "enable.auto.commit", "false");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+
+        testid = test_id_generate();
+        test_msgver_init(&mv, testid);
+
+        /* Produce messages */
+        topic = test_mk_topic_name("0137-barrier_batch_consume", 1);
+
+        for (p = 0; p < partition_cnt; p++)
+                test_produce_msgs_easy(topic, testid, p,
+                                       produce_msg_cnt / partition_cnt);
+
+        /* Create consumers */
+        consumer = test_create_consumer(topic, NULL, rd_kafka_conf_dup(conf), NULL);
+
+        test_consumer_subscribe(consumer, topic);
+        test_consumer_wait_assignment(consumer, rd_false);
+
+        /* Create generic consume queue */
+        rkq = rd_kafka_queue_get_consumer(consumer);
+
+        consumer_args.what             = "CONSUMER";
+        consumer_args.rkq              = rkq;
+        consumer_args.timeout_ms       = timeout_ms;
+        consumer_args.consume_msg_cnt  = consume_msg_cnt;
+        consumer_args.expected_msg_cnt = 4;
+        consumer_args.rk               = consumer;
+        consumer_args.testid           = testid;
+        consumer_args.mv               = &mv;
+        consumer_args.test             = test_curr;
+        if (thrd_create(&thread_id, consumer_batch_queue, &consumer_args) !=
+            thrd_success)
+                TEST_FAIL("Failed to create thread for %s", "CONSUMER");
+
+        parition_list = rd_kafka_topic_partition_list_new(1);
+        rd_kafka_topic_partition_list_add(parition_list, topic, pause_partition);
+
+        sleep(1);
+        err = rd_kafka_pause_partitions(consumer, parition_list);
+
+        if(err != RD_KAFKA_RESP_ERR_NO_ERROR) {
+            TEST_FAIL("Failed to pause partition %d for topic %s", pause_partition, topic);
+        }
+
+        sleep(1);
+
+        err = rd_kafka_resume_partitions(consumer, parition_list);
+
+        if(err != RD_KAFKA_RESP_ERR_NO_ERROR) {
+            TEST_FAIL("Failed to resume partition %d for topic %s", pause_partition, topic);
+        }
+
+        thrd_join(thread_id, NULL);
+
+
+        rd_kafka_message_t **abc =
```

Comment:
No variable declarations after statements, all varaibles declarations must go at the top of the scope.

Reply:
I am coming from a java background where it is advised to use variable declaration as near the first usage of it as possible. For my understanding, can you please tell me why is it important to have variable declaration at the top of the scope?

Reply:
Up until recently, the code wouldn't build on older MSVC compilers if variables were declared after statements, so that's why that's the current style in librdkafka.